### ISSUE TITLE
Respawn Anchor explosion protection is faulty

### ISSUE THREAD

**Issue by MithrandirCraft** [GriefPrevention/GriefPrevention#1115]
### What happened:
  BlockLandClaimExplosions: true
  BlockSurfaceCreeperExplosions: true
  BlockSurfaceOtherExplosions: false

Armor stands can be destroyed from outside claims using respawn anchor explosions.
If those can be destroyed, maybe something else can as well. But I think other blocks are somehow inmune.

### What was expected:
Armor stand should not be broken from any kind of explosion in LandClaims according to my configuration.

### Steps to reproduce:
1. Make a respawn anchor explode in overworld through vanilla procedure, close to an armor stand in a claim.
    Either within the claim or outside (a malicious player would do this outside)
2. Pick up the armorstand an anything on it.

### Server and GriefPrevention version:
Paste of `/version`: 
[21:31:32 INFO]: This server is running Paper version git-Paper-288 (MC: 1.16.4) (Implementing API version 1.16.4-R0.1-SNAPSHOT)
[21:31:32 INFO]: Checking version, please wait...
[21:31:32 INFO]: Previous version: git-Paper-262 (MC: 1.16.4)
[21:31:32 INFO]: You are running the latest version
Paste of `/version GriefPrevention`: 16.16.0

### Debugging steps attempted

- [ ] I attempted running only GriefPrevention on the server
  - [ ] Issue still occurs even if GriefPrevention is the only plugin running
- [ ] I attempted testing for the issue on a new server
  - [ ] Issue still occurs even after testing on a new server

**Comment by RoboMWM** [comment]
you can test with mobs if you want to be sure it really does affect more than armor stands

**Comment by Jikoo** [comment]
Damage by block explosions in general is not handled. This affects any living entity and should be able to be caused by beds as well.

**Comment by RoboMWM** [comment]
Interesting, thought EntityDamageEvent had something in there, but maybe it's only EntityDamagebyEntity

**Comment by Jikoo** [comment]
It's possible it used to handle block-based explosions properly - I've been wanting to get rid of [this check](https://github.com/TechFortress/GriefPrevention/blob/e8442fc5b1cf023c67b068424dbcb57dfc21f824/src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java#L919) made useless by [this line](https://github.com/TechFortress/GriefPrevention/blob/e8442fc5b1cf023c67b068424dbcb57dfc21f824/src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java#L754) for a while, but its existence could mean that GP used to handle other damage types differently.

**Comment by RoboMWM** [comment]
Yep, the infamous damage handler. Not sure if PvP protection should be entirely segregated to an addon or not, but that would be my first step in refactoring that mess.

**Comment by Jikoo** [comment]
I was planning on a refactor of it but ended up time crunched and lost steam. My start is [here](https://github.com/TechFortress/GriefPrevention/compare/master...Jikoo:dev/entity-damage-cleanup) - pretty much just extracting all of the logic into helper methods - but it's way too huge to be submitted as a single PR. My plan was to finish it up, then submit it split up by helper method for easy review. I really should get back to it, but I've been throwing all my free time into [having a ton of fun with Github Actions](https://github.com/lishid/OpenInv/pull/170) instead of actual Minecraft stuff.

In related news, I'm going to attempt to remember to follow [Conventional Commits spec](https://www.conventionalcommits.org/en/v1.0.0/) for the future to make things easier.

**Comment by RoboMWM** [comment]
I guess I am derailing, but github's discussions feature is _still_ in a closed beta. But hmm, has there ever been a standardized convention for commit messages before that? Also curious if github actions was just something you're looking at or if you saw me commit some stuff earlier this week lol.

Back to the issue, interesting tho; I remember an issue regarding a tnt minecart killing animals. And... hmm, tnt itself turns into an entity. So I guess being able to trigger block explosions in the overworld is something relatively new?

**Comment by Jikoo** [comment]
Continuing derail: Yes and no. Interestingly, Conventional Commits goes directly against [others' recommendations](https://github.com/joelparkerhenderson/git_commit_message#reject-these-formats) - seems like it spawned from the Angular community, and depending on who you ask it may or may not be horrible. People often recommend (and it seems like the route GitHub themselves take) just using [keywords](https://github.com/joelparkerhenderson/git_commit_message#summary-keywords) for the first word of a commit. I had been looking into Actions for a while because I was very interested in setting up a CI environment that would allow me to (legally) build against NMS without having to host something myself or encrypt files. I brought it up because I saw the inclusion of release-drafter - I know that one uses labels on PRs, but I saw several alternatives that parse commit messages for a more complete changelog.

Re: issue - yeah, it should only have been possible with beds in the nether and end before, everything else converts to/is already an entity prior to detonation. This should be a simple fix, I'll look into it.

**Comment by Jikoo** [comment]
Not exactly simple after all - turns out Bukkit doesn't quite offer enough information to us to fully guard against block explosions. Going to PR what I've got and discuss the shortcomings there.
